src/pages/CreateReviewPage.js

```diff
@@ -0,0 +1,158 @@
+  const handleReviewSubmit = async ({ title, content }) => {
+    setErrorMessage(undefined);
+    setIsLoading(true);
+
+    if (ratingValue === 0) {
+      setRatingError(true);
+      setErrorMessage('Avaliação em falta.');
+      setIsLoading(false);
+      return;
+    }
+    setRatingError(false);
+
+    try {
+      const requestBody = {
+        title,
+        content,
+        rating: ratingValue,
+        userId: user._id,
+        orderId,
+      };
+
+      let response = await createReview(requestBody);
+      console.log(response.data);
+
+      setSuccessMessage('Review criado com sucesso. Obrigado =)');
+    } catch (error) {
+      const errorDescription = error.response.data.message;
+      setErrorMessage(errorDescription);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <Box sx={addReviewPageClasses.container}>
+      <Typography variant={componentProps.variant.h2} color={componentProps.color.primary} sx={{ my: 4 }}>
+        REVIEW
+      </Typography>
+      {order && !successMessage && (
+        <Box sx={addReviewPageClasses.form}>
+          <ShopOrder order={order} />
+          <form onSubmit={handleSubmit(handleReviewSubmit)} noValidate>
+            <Box sx={ratingError ? addReviewPageClasses.ratingError : addReviewPageClasses.rating} align='left'>
+              <Typography component='legend'>Avaliação</Typography>
+              <Rating name='rating-controlled' onChange={handleRatingChange} value={ratingValue} />
+            </Box>
+            <Controller
+              name={componentProps.name.title}
+              control={control}
+              rules={{ required: 'Titulo em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Titulo'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={addReviewPageClasses.formField}
+                  error={!!errors.title}
+                  autoComplete='true'
+                  autoFocus
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              name={componentProps.name.content}
+              control={control}
+              rules={{ required: 'Descrição em falta', minLength: { value: 30, message: 'Avaliação deve conter pelo menos 25 caracteres.' } }}
+              render={({ field }) => (
+                <TextField label='Descrição' maxRows={4} multiline sx={addReviewPageClasses.formTextArea} placeholder='Escreva aqui a descrição...' error={!!errors.content} {...field} />
+              )}
+            />
+
+            <button type={componentProps.type.submit} ref={submitBtnRef} hidden>
+              Enviar
+            </button>
+          </form>
+        </Box>
+      )}
+
+      {successMessage && <SuccessMessage message={successMessage} />}
+      {errorMessage && <ErrorMessage message={errorMessage} />}
```

Comment:
You can use react-hook-form values to display success or error messages instead of create a new state for that. Take a look in isSubmitted and errors in react-hook-form: https://react-hook-form.com/api/useform/formstate/#return

Reply:
This `ErrorMessage` state is more to display server errors that may occur. If i would use the one from RHF it would be attached only to the form, right?